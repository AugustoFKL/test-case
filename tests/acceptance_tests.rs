#![cfg(test)]

mod acceptance {
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout).to_string();
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout).to_string();
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        if !env::var("TEST_CASE_ACCEPTANCE_RUN_ASYNC").map_or(true, |v| v == "true") {
            return; // Test can be skipped via environment variable, when ran on rust < 1.41.0
        }

        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout).to_string();
        insta::assert_display_snapshot!(lines);
    }
}
