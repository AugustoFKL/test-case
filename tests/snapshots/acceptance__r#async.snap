---
source: tests/acceptance_tests.rs
expression: lines
---

running 2 tests
test test_cases::works_seamlessly_with_tokio::_100i32_expects_100usize ... ok
test test_cases::works_seamlessly_with_async_std::_100i32_expects_100usize ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


